pallets/treasury/src/lib.rs

```diff
@@ -304,23 +305,54 @@ pub mod pallet {
 
             // count actors
             let (count_devs, count_execs) =
-                <Beneficiaries<T>>::iter_keys().fold((0, 0), |mut acc, c| {
+                <Beneficiaries<T>>::iter_keys().fold((0, 0), |mut acc: (i64, i64), c| {
                     match c.1 {
                         BeneficiaryRole::Developer => {
-                            acc.0 += 1;
+                            if let Some(v) = acc.0.checked_add(1) {
+                                acc.0 = v
+                            }
+                            // cannot return Err
+                            // else {
+                            //     return Err(Error::ArithmeticError);
+                            // }
                         },
                         BeneficiaryRole::Executor => {
-                            acc.1 += 1;
+                            if let Some(v) = acc.1.checked_add(1) {
+                                acc.1 = v
+                            }
+                            // cannot return Err
+                            // else {
+                            //     return Err(Error::ArithmeticError);
+                            // }
                         },
                     }
                     acc
                 });
 
             // calculate relative rewards per actor
-            let relative_per_dev =
-                Perbill::from_rational(1, count_devs as u32) * inflation_alloc.developer;
-            let relative_per_exec =
-                Perbill::from_rational(1, count_execs as u32) * inflation_alloc.executor;
+            let relative_per_dev = if let Some(v) =
+                Perbill::from_rational(1, count_devs as u32).checked_mul(&inflation_alloc.developer)
+            {
+                v
+            } else {
+                Perbill::from_rational(1, count_devs as u32)
+            };
+            // cannot return Err
```

Comment:
create todos and open corresponding issue

Reply:
Added the TODOs and the issue